src/DealClient.sol

```diff
@@ -28,32 +19,32 @@ contract DealClient {
         owner = msg.sender;
     }
 
-    function addCID(bytes calldata cidraw, uint size) public {
+    function addCID(bytes calldata pieceCID, uint size) public {
        require(msg.sender == owner);
-       cidSet[cidraw] = true;
-       cidSizes[cidraw] = size;
+       cidSet[pieceCID] = true;
+       //cidSizes[pieceCID] = size;
     }
 
-    function policyOK(bytes calldata cidraw, bytes calldata provider) internal view returns (bool) {
-        bool alreadyStoring = cidProviders[cidraw][provider];
+    function policyOK(bytes calldata pieceCID, bytes calldata provider) internal view returns (bool) {
+        bool alreadyStoring = cidProviders[pieceCID][provider];
         return !alreadyStoring;
     }
 
-    function authorizeData(bytes calldata cidraw, bytes calldata provider, uint size) public {
+    function authorizeData(bytes calldata pieceCID, bytes calldata provider, uint size) public {
         // if (msg.sender != f05) return;
-        require(cidSet[cidraw], "cid must be added before authorizing");
-        require(cidSizes[cidraw] == size, "data size must match expected");
-        require(policyOK(cidraw, provider), "deal failed policy check: has provider already claimed this cid?");
+        require(cidSet[pieceCID], "cid must be added before authorizing");
+        require(cidSizes[pieceCID] == size, "data size must match expected");
+        require(policyOK(pieceCID, provider), "deal failed policy check: has provider already claimed this cid?");
 
-        cidProviders[cidraw][provider] = true;
+        cidProviders[pieceCID][provider] = true;
     }
 
     function handle_filecoin_method(uint64 method, uint64, bytes calldata params) public {
         // dispatch methods
         if (method == AUTHORIZE_MESSAGE_METHOD_NUM) {
             bytes calldata deal_proposal_cbor_bytes = specific_authenticate_message_params_parse(params);
-            (bytes calldata cidraw, bytes calldata provider, uint size) = specific_deal_proposal_cbor_parse(deal_proposal_cbor_bytes);
-            authorizeData(cidraw, provider, size);
+            (bytes calldata pieceCID, bytes calldata provider, uint size) = specific_deal_proposal_cbor_parse(deal_proposal_cbor_bytes);
```

Comment:
Id suggest keeping specific_deal_proposal returning size, just don't pass it into authorize data